src/misc/messages.cpp

```diff
@@ -204,14 +249,17 @@ void MSG_Init([[maybe_unused]] Section_prop *section)
 		return;
 	}
 
-	// If a short-hand name was provided then add the file extension
-	const auto lng_file = lang + (ends_with(lang, ".lng") ? "" : ".lng");
+	bool result = false;
+	if (ends_with(lang, ".lng"))
+		result = load_message_file(GetResourcePath("translations", lang));
+	else
+		// If a short-hand name was provided then add the file extension
+		result = load_message_file(GetResourcePath("translations", lang + ".lng"));
```

Comment:
Extract "translations" and ".lng" to constexpr?

Reply:
Only `static const` was possible for now; it seems we need C++20 for `constexpr` here.